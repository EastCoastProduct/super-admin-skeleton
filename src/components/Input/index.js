import React, { PropTypes } from 'react';

const Input = (props) => {
  const { label, input, meta: { active, touched, error }, ...other } = props;

  return (
    <div>
      <label htmlFor={input.name}>
        {label}
        <input id={input.name} {...input} {...other} />
      </label>
      {!active && touched && error && <p>{error}</p>}
    </div>
  );
};

Input.propTypes = {
  label: PropTypes.string,
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
};

export default Input;
