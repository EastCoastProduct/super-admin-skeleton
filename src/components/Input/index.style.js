import { StyleSheet } from 'aphrodite/no-important';
import variables from '../../styles/variables';

const { colors, sizes } = variables;

export default StyleSheet.create({
  input: {
    background: colors.input,
    border: `1px solid ${colors.header}`,
    borderRadius: sizes.radius,
    boxShadow: `inset 1px 1px ${colors.shadow}`,
    color: colors.font,
    height: sizes.inputHeight,
    padding: '5px 10px',
    width: '100%',
  },
  label: {
    color: colors.header,
    display: 'inline-block',
    fontWeight: 700,
    lineHeight: `${sizes.inputHeight}px`,
    marginRight: 20,
    textTransform: 'uppercase',
    verticalAlign: 'top',
  },
  holder: {
    display: 'inline-block',
    marginBottom: 50,
    verticalAlign: 'top',
    width: '78%',
  },
  holderErr: {
    marginBottom: 0,
  },
});
