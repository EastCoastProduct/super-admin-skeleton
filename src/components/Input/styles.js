import { StyleSheet } from 'aphrodite/no-important';
import { colors, sizes } from '../../styles/variables';

const shared = {
  input: {
    background: colors.input,
    border: `1px solid ${colors.header}`,
    borderRadius: sizes.radius,
    boxShadow: `inset 1px 1px ${colors.shadow}`,
    color: colors.font,
    padding: '9px 10px',
    width: '100%',
  },
};

export default StyleSheet.create({
  holder: {
    marginBottom: 20,
  },
  input: {
    ...shared.input,
    height: sizes.inputHeight,
  },
  textarea: {
    ...shared.input,
    minHeight: sizes.textareaHeight,
    resize: 'vertical',
  },
  validatedHolder: {
    marginBottom: 50,
  },
  errHolder: {
    marginBottom: 15,
  },
  label: {
    color: colors.header,
    display: 'block',
    fontWeight: 700,
    marginBottom: 5,
    textAlign: 'left',
    textTransform: 'uppercase',
  },
  error: {
    margin: '5px 0 0',
  },
});
