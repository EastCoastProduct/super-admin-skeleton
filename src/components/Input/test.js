import React from 'react';
import { shallow } from 'enzyme';
import deepMerge from 'deepmerge';
import Input, { isError } from './';

describe('Input component snapshot', () => {
  const props = {
    id: 'Login',
    input: {
      name: 'email',
    },
    meta: {
      active: false,
      error: false,
      touched: false,
    },
  };
  const extraProps = deepMerge(props, {
    label: 'Email',
    type: 'email',
    validated: true,
  });

  it('renders basic required data', () => {
    const newProps = deepMerge(props, {
      type: 'email',
    });
    const wrapper = shallow(<Input {...newProps} />);

    expect(wrapper).toMatchSnapshot();
    expect(isError(newProps.meta)).toBeFalsy();
  });

  it('renders label and is focused', () => {
    const newProps = deepMerge(props, {
      id: 'EditUser',
      input: {
        name: 'firstname',
        value: 'John',
      },
      label: 'First Name',
      meta: {
        active: true,
        touched: true,
      },
    });
    const wrapper = shallow(<Input {...newProps} />);

    expect(wrapper).toMatchSnapshot();
    expect(isError(newProps.meta)).toBeFalsy();
  });

  it('renders label and input is required', () => {
    const newProps = deepMerge(extraProps, {
      meta: {
        error: 'Input required',
        touched: true,
      },
    });
    const wrapper = shallow(<Input {...newProps} />);

    expect(wrapper).toMatchSnapshot();
    expect(isError(newProps.meta)).toBeTruthy();
  });

  it('renders label and input is invalid', () => {
    const newProps = deepMerge(extraProps, {
      input: {
        value: 'notAnEmail',
      },
      meta: {
        error: 'Input is invalid',
        touched: true,
      },
    });
    const wrapper = shallow(<Input {...newProps} />);

    expect(wrapper).toMatchSnapshot();
    expect(isError(newProps.meta)).toBeTruthy();
  });

  it('renders extra info and it is active and invalid', () => {
    const newProps = deepMerge(extraProps, {
      input: {
        value: 'notAnEmail',
      },
      meta: {
        active: true,
        error: 'Input is invalid',
        touched: true,
      },
      placeholder: 'Email',
    });
    const wrapper = shallow(
      <Input {...newProps} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(isError(newProps.meta)).toBeFalsy();
  });

  it('renders as textarea with max length', () => {
    const newProps = deepMerge(props, {
      input: {
        name: 'bio',
        value: 'This is my bio',
      },
      label: 'Bio',
      maxLength: '1000',
      meta: {
        touched: true,
      },
      placeholder: 'Bio',
      textarea: true,
    });
    const wrapper = shallow(
      <Input {...newProps} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(isError(newProps.meta)).toBeFalsy();
  });
});
