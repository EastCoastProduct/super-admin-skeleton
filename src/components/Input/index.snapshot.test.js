import React from 'react';
import renderer from 'react-test-renderer';
import Input from './';

describe('Input component snapshot', () => {
  it('renders basic required data', () => {
    const meta = {
      active: false,
      error: false,
      touched: false,
    };
    const input = {
      name: 'email',
      value: '',
    };
    const tree = renderer.create(<Input meta={meta} input={input} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders label and is focused', () => {
    const meta = {
      active: true,
      error: false,
      touched: true,
    };
    const input = {
      name: 'email',
      value: 'test@email.com',
    };
    const tree = renderer.create(
      <Input meta={meta} input={input} label="Email" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders label and input is required', () => {
    const meta = {
      active: false,
      error: 'Input required',
      touched: true,
    };
    const input = {
      name: 'email',
      value: '',
    };
    const tree = renderer.create(
      <Input meta={meta} input={input} label="Email" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders label and input is invalid', () => {
    const meta = {
      active: false,
      error: 'Input is invalid',
      touched: true,
    };
    const input = {
      name: 'email',
      value: 'notAnEmail',
    };
    const tree = renderer.create(
      <Input meta={meta} input={input} label="Email" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders extra info and it is active and invalid', () => {
    const meta = {
      active: true,
      error: 'Input is invalid',
      touched: true,
    };
    const input = {
      name: 'email',
      value: 'notAnEmail',
    };
    const tree = renderer.create(
      <Input
        meta={meta}
        input={input}
        label="Email"
        type="email"
        placeholder="Email"
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
