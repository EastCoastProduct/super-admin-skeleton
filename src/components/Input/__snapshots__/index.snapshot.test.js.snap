exports[`Input component snapshot renders basic required data 1`] = `
<div>
  <label
    htmlFor="email">
    <input
      id="email"
      name="email"
      value="" />
  </label>
</div>
`;

exports[`Input component snapshot renders extra info and it is active and invalid 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <input
      id="email"
      name="email"
      placeholder="Email"
      type="email"
      value="notAnEmail" />
  </label>
</div>
`;

exports[`Input component snapshot renders label and input is invalid 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <input
      id="email"
      name="email"
      value="notAnEmail" />
  </label>
  <p>
    Input is invalid
  </p>
</div>
`;

exports[`Input component snapshot renders label and input is required 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <input
      id="email"
      name="email"
      value="" />
  </label>
  <p>
    Input required
  </p>
</div>
`;

exports[`Input component snapshot renders label and is focused 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <input
      id="email"
      name="email"
      value="test@email.com" />
  </label>
</div>
`;
